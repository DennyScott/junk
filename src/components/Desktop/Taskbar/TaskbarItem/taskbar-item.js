import React from 'react';
import PropTypes from 'prop-types';
import './taskbar-item.css';

const taskbarItem = props =>(
    <div className="taskbar-item">
        {props.program.name}
    </div>
);

taskbarItem.PropTypes = {
    program: PropTypes.object.isRequired,
}

export default taskbarItem;