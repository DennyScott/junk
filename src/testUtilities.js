import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export const rendersCorrectly =  component => {
  const wrapper = shallow(component);
  expect(wrapper.length).toBe(1);
}

export const matchesSnapshot = component => {
  const wrapper = shallow(component);
  expect(toJson(wrapper)).toMatchSnapshot();
}