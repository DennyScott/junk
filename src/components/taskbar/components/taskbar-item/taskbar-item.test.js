import React from 'react';
import TaskbarItem from './taskbar-item';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { rendersCorrectly, matchesSnapshot } from 'testUtilities';

const minProps = {
    name: '',
    onClick: jest.fn(),
};

const getMinComponent = otherProps => <TaskbarItem {...minProps} {...otherProps} />;

it('TaskbarItem renders correctly', () => {
    rendersCorrectly(getMinComponent(), expect);
});

it('TaskbarItem render matches snapshot', () => {
    matchesSnapshot(getMinComponent(), expect);
});

it('Name gets displayed in taskbar', () => {
    const wrapper = shallow(getMinComponent({ name: 'test' }));
    expect(toJson(wrapper.find('.taskbar-item-name').children())).toBe('test');
});

it('taskbar item click will call click function', () => {
    var clickFunc = jest.fn();
    const wrapper = shallow(getMinComponent({ onClick: clickFunc }));
    wrapper.simulate('click');
    expect(clickFunc).toHaveBeenCalled();
});
