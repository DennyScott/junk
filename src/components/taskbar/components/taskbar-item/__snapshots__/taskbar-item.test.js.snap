// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskbarItem render matches snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.img />
  <styled.span
    className="taskbar-item-name"
  />
</styled.div>
`;
